x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -204,6 +207,7 @@ export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =
           ),
           threshold: mapToConditionsLookup(criteria, (c) => c.threshold),
           metric: mapToConditionsLookup(criteria, (c) => c.metric),
+          ...additionalContext,
```

Comment:
We shouldn't spread `additionalContext` like this since it is also adding `currentValue`, `isNoData`, `isError` and whatever else is part of the `ConditionResult` object.  I think the additional context should be added  to the `ConditionResult` object as an additional field like:

```
type ConditionResult = InventoryMetricConditions & {
  shouldFire: boolean;
  shouldWarn: boolean;
  currentValue: number;
  isNoData: boolean;
  isError: boolean;
  context: AdditionalContext;
}
```
then change line 197 to
```
const additionalContext = results && results.length > 0 ? results[0][group].context : null;
```
For reference, I discovered this by creating a server log action with `{{context}}` as the message to log to the Kibana logs; that created this object for my laptop:

```JSON
{
    "alertState": "ALERT",
    "comparator": ">",
    "currentValue": 0.18328125,
    "customMetric": {
        "aggregation": "avg",
        "field": "",
        "id": "alert-custom-metric",
        "type": "custom"
    },
    "group": "2019-Macbook-Pro.local",
    "host": {
        "architecture": "x86_64",
        "hostname": "2019-Macbook-Pro.local",
        "id": "62A34270-7062-5A9B-A3BD-E7553999C9FF",
        "ip": [
            "fe80::aede:48ff:fe00:1122",
            "fe80::448:ebac:5e82:d8ac",
            "2601:285:8880:510:c57:4f19:84df:788d",
            "2601:285:8880:510::fb3b",
            "fdd1:910a:985b:4d6b:1cf0:1b30:6cdf:9b9d",
            "2601:285:8880:510:1c19:fb8f:3078:2961",
            "2601:285:8880:510:f178:70f9:5ce6:f754",
            "2601:285:8880:510:9119:9930:60c3:66b",
            "2601:285:8880:510:64dd:1978:5228:b1b1",
            "2601:285:8880:510:dd5d:b426:ab83:8316",
            "2601:285:8880:510:4199:8f85:a97b:acd5",
            "10.0.0.124",
            "2601:285:8880:510:ccf5:4824:8f5b:4aea",
            "fe80::7c00:b6ff:fe11:d406",
            "fe80::7c00:b6ff:fe11:d406",
            "fe80::dfca:af6d:eee1:18c1",
            "fe80::5739:8bd0:f4de:e215",
            "fe80::aaa9:6c99:9944:aad4",
            "fe80::4d73:70d0:fdd8:9a32",
            "fe80::ad0b:df5b:791d:2271",
            "fe80::1ca7:b7fb:1c72:eca2"
        ],
        "mac": [
            "00-E0-4C-68-01-A9",
            "3C-22-FB-53-75-19",
            "3E-22-FB-53-75-19",
            "7E-00-B6-11-D4-06",
            "82-4D-9E-49-5C-00",
            "82-4D-9E-49-5C-01",
            "82-4D-9E-49-5C-04",
            "82-4D-9E-49-5C-05",
            "AC-DE-48-00-11-22"
        ],
        "name": "2019-Macbook-Pro.local",
        "os": {
            "build": "20G527",
            "family": "darwin",
            "kernel": "20.6.0",
            "name": "macOS",
            "platform": "darwin",
            "type": "macos",
            "version": "11.6.5"
        }
    },
    "isError": false,
    "isNoData": false,
    "metric": "cpu",
    "reason": "CPU usage is 18.3% in the last 1 min for 2019-Macbook-Pro.local. Alert when > 1%.",
    "shouldFire": true,
    "shouldWarn": false,
    "threshold": [
        1
    ],
    "timeSize": 1,
    "timeUnit": "m",
    "timestamp": "2022-10-04T16:28:43.336Z",
    "value": {
        "condition0": "18.3%"
    },
    "viewInAppUrl": "/app/metrics/link-to/inventory?customMetric=&metric=%28type%3Acpu%29&nodeType=host&timestamp=1664900923336"
}
```

That's when I noticed the additional fields were being added.

Reply:
You are right. It was also adding other fields that are not part of the context. I have modified the code now.